src/InputFilter/RestService/PatchInputFilter.php

```diff
@@ -75,7 +75,7 @@ public function init()
             'continue_if_empty' => true,
             'validators'        => [
                 new CallbackValidator(function ($value) {
-                    if (is_numeric($value) && intval($value) !== $value) {
+                    if (!is_numeric($value) || intval($value) != $value) {
```

Comment:
This can be fixed without juggling types implicitly:

```php
if (! is_numeric($value) || ((string) $value) !== (string) (int) $value) {
```

Yes, I know the AST is a bit more complex, but the implicits of `!=` keep changing in PHP, so a strict comparison is preferred anyway.